### ISSUE TITLE
Show challenge - missing challengeInfoText

### ISSUE THREAD

**Issue by kaserdan** [Adyen/adyen-3ds2-android#26]
Hi, 
I am getting an error trying to show challenge:
```
Error [code: 201, description: Required Data Element Missing., details: Invalid JSON for NativeChallenge. No value for challengeInfoText]
```
We have notified card issuer about this missing field and we got sample CRes with their data elements, which they consider to be valid:

```
{
  "threeDSServerTransID" : "c92a4ee7-1ae7-45ab-aca1-5451b5dcb430",
  "acsCounterAtoS" : "000",
  "acsTransID" : "bac4f581-80a5-41d2-9b36-ec3134fd60bc",
  "acsUiType" : "04",
  "challengeAddInfo" : "Potvrďte platbu v aplikaci George klíč ve Vašem mobilním telefonu, který má přístup na internet.\nDetail platby:\n Alza.cz a.s.\n 9,00 CZK\n 4.2.2021 18:55\n XX5065",
  "challengeCompletionInd" : "N",
  "challengeInfoHeader" : "Potvrzení platby kartou",
  "expandInfoLabel" : "O zabezpečení karty na internetu",
  "expandInfoText" : "Služba 3D Secure chrání Vaši platební kartu před zneužitím. Každou transakci musíte před odesláním ověřit dohodnutým způsobem. Transakce proběhne až po tomto ověření.",
  "issuerImage" : {
    "medium" : "https://3dsecure2.csas.cz/acsas/static/csas/img/csas_medium.png",
    "high" : "https://3dsecure2.csas.cz/acsas/static/csas/img/csas_high.png",
    "extraHigh" : "https://3dsecure2.csas.cz/acsas/static/csas/img/csas_extraHigh.png"
  },
  "messageVersion" : "2.1.0",
  "oobContinueLabel" : "Pokračovat",
  "psImage" : {
    "medium" : "https://3dsecure2.csas.cz/acsas/static/common/img/mastercard_medium.png",
    "high" : "https://3dsecure2.csas.cz/acsas/static/common/img/mastercard_high.png",
    "extraHigh" : "https://3dsecure2.csas.cz/acsas/static/common/img/mastercard_extraHigh.png"
  },
  "sdkTransID" : "114d41c3-c5cc-42a8-b78e-eecaa5ee5089",
  "whyInfoLabel" : "Nápověda",
  "whyInfoText" : "Podrobnější informace získáte na www.csas.cz nebo na informační lince 800 207 207 (ze zahraničí volejte +420 956 777 901). Jste-li držitelem služební platební karty, volejte +420 956 777 888.",
  "messageType" : "CRes"
}
```

Which could be valid UI template as shown in [EMV® 3-D Secure Protocol and Core Functions Specification](https://www.emvco.com/document-search/?action=search_documents&publish_date=&emvco_document_version=2-1-0&emvco_document_book=&px_search=&emvco_document_technology%5B%5D=3-d-secure), Figure 4.12, page 86.
This doc also notes following for `challengeAddInfo`:

> Text provided by the ACS/Issuer to Cardholder during OOB authentication to **replace** Challenge Information Text and Challenge Information Text Indicator in the OOB Template.

Is `challengeInfoText` really required in this case?

**Comment by mohammedDehairy** [comment]
Hi @kaserdan ,

I am afraid they misinterpreted the specs, `challengeAddInfo` doesn't replace `challengeInfoText` completely, `challengeAddInfo` replaces `challengeInfoText` only after the shopper leave the merchant app to do the OOB challenge and then come back to the app, but before the shopper leaves the merchant app the `challengeInfoText` is used and required.

As mentioned in the `Protocol and Core Functions Specification` section `3.2`:

```
When a Cardholder returns to the 3DS Requestor App from another app, the ACS can also utilise the Challenge Additional Info Text element (for the Native UI) or the ACS HTML Refresh element (for the HTML UI) to improve the UI experience. In this scenario, the SDK will automatically display these data elements when the 3DS Requestor App is moved to the foreground.
```

So `challengeAddInfo` replaces `challengeInfoText` only during an OOB challenge and after the app goes to background and back again.

Mohamed

**Comment by kaserdan** [comment]
Hi Mohamed,
Unfortunatly this issue is still present for this specific card issuer. Do you have any suggestion on how to encourage them into fixing this? 
Otherwise we have to downgrade to version 2.1.0-rc09.

**Comment by caiofaustino** [comment]
I'll close this since it has been parked for a while. Feel free to reopen if you still have problems.

**Comment by Gp2mv3** [comment]
Hello @kaserdan,

I've exactly the same issue with field `challengeInfoLabel` for a specific card issuer.
Did you manage to do something on your side to avoid the issues or should we just contact the issuer ?

Thaks !

**Comment by kaserdan** [comment]
Hi, unfortunately there was no other way. We had contacted the issuer and halted the version bump until they deployed fix on their side.
